src/Sulu/Bundle/SecurityBundle/Resources/public/js/components/roles/edit/main.js

```diff
@@ -0,0 +1,155 @@
+            return RoleManager.loadOrNew(this.options.id);
+        },
+
+        header: function() {
+            var config =  {
+                tabs: {
+                    url: '/admin/content-navigations?alias=roles',
+                    options: {
+                        data: function() {
+                            return this.data;
+                        }.bind(this)
+                    },
+                    componentOptions: {
+                        values: this.data
+                    }
+                },
+                toolbar: {
+                    buttons: {
+                        save: {
+                            parent: 'saveWithOptions'
+                        }
+                    }
+                }
+            };
+
+            if (!!this.options.id) {
+                config.toolbar.buttons.delete = {};
+            }
+
+            return config;
+        },
+
+        initialize: function() {
+            this.bindCustomEvents();
+        },
+
+        bindCustomEvents: function() {
+            this.sandbox.on('sulu.header.back', RoleRouter.toList);
+            this.sandbox.on('sulu.toolbar.delete', this.deleteRole.bind(this));
+            this.sandbox.on('sulu.tab.dirty', this.enableSave.bind(this));
+            this.sandbox.on('sulu.router.navigate', this.disableSave.bind(this));
+            this.sandbox.on('sulu.toolbar.save', this.save.bind(this));
+            this.sandbox.on('sulu.tab.save-error', this.disableSave.bind(this));
+            this.sandbox.on('sulu.tab.saving', this.loadingSave.bind(this));
+            this.sandbox.on('sulu.tab.data-changed', this.changeData.bind(this));
+        },
+
+        changeData: function(data) {
+            this.data = data;
+        },
+
+        enableSave: function() {
+            this.sandbox.emit('sulu.header.toolbar.item.enable', 'save', false);
+        },
+
+        disableSave: function() {
+            this.sandbox.emit('sulu.header.toolbar.item.disable', 'save', false);
+        },
+
+        loadingSave: function() {
+            this.sandbox.emit('sulu.header.toolbar.item.loading', 'save');
+        },
+
+        /**
+         * Asks the user if the role should be deleted. When confirmed
+         * The role gets deleted and the application navigates to the list.
+         */
+        deleteRole: function() {
+            this.sandbox.sulu.showDeleteDialog(function(wasConfirmed) {
+                if (wasConfirmed) {
+                    RoleManager.delete(this.options.id).then(function() {
+                        RoleRouter.toList();
+                    }.bind(this));
+                }
+            }.bind(this));
+        },
+
+        /**
+         * Saves the current tab.
+         *
+         * @returns {Object} a promise.
+         */
+        saveTab: function() {
+            var promise = $.Deferred();
+            this.sandbox.once('sulu.tab.saved', function(savedData, updateData) {
```

Comment:
Why are you registering that event using `once`? If we simply do that in `bindCustomEvents` with `on` it would probably also work. Without the risk that the event handler is executed multiple times, in case it is not called later.